dask/dataframe/io/parquet/arrow.py

```diff
@@ -119,6 +120,51 @@ def _determine_dataset_parts(fs, paths, gather_statistics, filters, dataset_kwar
     return parts, dataset
 
 
+def _write_partitioned(table, root_path, partition_cols, filesystem=None, **kwargs):
+    """ Write table to a partitioned dataset with pyarrow.
+
+        Logic copied from pyarrow.parquet.
+        (arrow/python/pyarrow/parquet.py::write_to_dataset)
+    """
+    fs, root_path = pq._get_filesystem_and_path(filesystem, root_path)
+    pq._mkdir_if_not_exists(fs, root_path)
+
+    df = table.to_pandas(ignore_metadata=True)
+    partition_keys = [df[col] for col in partition_cols]
+    data_df = df.drop(partition_cols, axis="columns")
+    data_cols = df.columns.drop(partition_cols)
+    if len(data_cols) == 0:
+        raise ValueError("No data left to save outside partition columns")
+
+    subschema = table.schema
+    for col in table.schema.names:
+        if col in partition_cols:
+            subschema = subschema.remove(subschema.get_field_index(col))
+
+    md_list = []
+    for keys, subgroup in data_df.groupby(partition_keys):
+        if not isinstance(keys, tuple):
+            keys = (keys,)
+        subdir = "/".join(
+            [
+                "{colname}={value}".format(colname=name, value=val)
+                for name, val in zip(partition_cols, keys)
+            ]
+        )
+        subtable = pa.Table.from_pandas(
+            subgroup, preserve_index=False, schema=subschema, safe=False
+        )
+        prefix = "/".join([root_path, subdir])
+        pq._mkdir_if_not_exists(fs, prefix)
+        outfile = guid() + ".parquet"
+        full_path = "/".join([prefix, outfile])
+        with fs.open(full_path, "wb") as f:
+            pq.write_table(subtable, f, metadata_collector=md_list, **kwargs)
+        md_list[-1].set_file_path("/".join([subdir, outfile]))
```

Comment:
@jorisvandenbossche - We can go back to using pyarrow's `write_to_dataset` api if that function can set the file path for new metadata objects, or pass back the appropriate paths.  Do you have any thoughts/suggestions on this?

Reply:
Yes, at least we should also return the path (so it can be fixed afterwards), or otherwise set it ourselves. Starting to also return the path is not really backwards compatible. About changing the path, that might depend on whether there are other use cases for collecting this metadata (although I *assume* dask is the main user of this keyword). In any case worth opening a JIRA issue.

Reply:
Opened [ARROW-8244](https://issues.apache.org/jira/projects/ARROW/issues/ARROW-8244) to discuss/address this.